### DISCUSSION TITLE
Multi-cluster GitRepository authentication

### DISCUSSION THREAD

**Discussion by mimmus** [fluxcd/flux2#3110] [Q&A] [answered]
I have a Multi-cluster infrastructure with Cluster API: I'm going to use Flux to manage apps both in the management and in workload clusters.
If I define an authenticated GitRepository with credentials in a k8s secret, these are not "propagated" to managed clusters and in fact this is the Gitrepository status:

```
NAME           URL                                                     AGE     READY   STATUS
example-repo   http://gitcs01.mydomain.com/devops/d2iq/gitops.git   5h22m   False   failed to get secret 'dev-ql2mz-hns4l/gitlab-creds': Secret "gitlab-creds" not found
```

What is the best practice to do this?

**Comment by stefanprodan** [comment, marked as answer]
The secret could be placed in Git encrypted with SOPS, from there Flux running on the management cluster would connect to the leaf clusters and reconcile the secret there. Some docs around this:
- https://fluxcd.io/flux/components/kustomize/kustomization/#secrets-decryption
- https://fluxcd.io/flux/components/kustomize/kustomization/#remote-clusters--cluster-api

**Comment by mimmus** [comment]
Thank you very much, I'm totally unaware of all of these interesting considerations about security in GitOps.
World is becoming always more difficult for an old-seasoned sysadmin!
